edgelet/edgelet-http/src/util/hyperwrap.rs

```diff
@@ -46,6 +49,21 @@ impl Config {
     }
 }
 
+fn uri_to_proxy(uri: Uri) -> Result<Proxy, Error> {
+    let cloned = uri.clone();
```

Comment:
You don't use `uri` except for line 54, so if you move this down after line 54 you wouldn't need to `clone()` it?